### PR TITLE
Updated broken links.

### LINKED ISSUE #2655: Depreciated link in CONTRIBUTING.md
In the [Workflow](https://github.com/pybamm-team/PyBaMM/blob/d218aa45dc20f3f7e1ba8cfaf61fcd09763ebcdc/CONTRIBUTING.md#workflow) of CONTRIBUTING.md
>4. [Install](https://pybamm.readthedocs.io/en/latest/install/install-from-source.html) PyBaMM with the developer options.

Points to a depreciated installation link.

The correct link would be https://pybamm.readthedocs.io/en/latest/source/user_guide/installation/install-from-source.html

### LINKED ISSUE #2660: Fix broken URLs
URL checker was down temporarily but #2659 brings it back up again.

At the moment, there are a lot of broken URLs - https://github.com/pybamm-team/PyBaMM/actions/runs/4090861650/jobs/7054568267 - which should be fixed.

Most of them originate from the documentation revamp, some of them are false alarms, and some of them are from external repositories/sources.

### PR DESCRIPTION
**Pull request by jeromtom**
# Description
Changes the depreciated/broken link in CONTRIBUTING.md to the correct working link:
https://pybamm.readthedocs.io/en/latest/source/user_guide/installation/install-from-source.html

Fixes #2655 and partially fixes #2660

## Type of change
Documentation Update
- [X] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [X] No style issues: `$ pre-commit run`
- [X] All tests pass: `$ python run-tests.py --unit`
- [X] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 - Fixed a bug where the solid phase conductivity was double-corrected for tortuosity when loading parameters from a BPX file ([#2638](https://github.com/pybamm-team/PyBaMM/pull/2638)).
 - Changed termination from "success" to "final time" for algebraic solvers to match ODE/DAE solvers ([#2613](https://github.com/pybamm-team/PyBaMM/pull/2613)).
+- Changed depreciated installation link in CONTRIBUTING.md ([#2656](https://github.com/pybamm-team/PyBaMM/pull/2656)).
```
Comment (Saransh-cpp) [inline CHANGELOG.md:16]: I don't think this requires a changelog entry

**Review by Saransh-cpp — COMMENTED** [review]
Thanks, @jeromtom! Will you be pushing more URL fixes to this PR?

File: CHANGELOG.md (hunk 1 above)
Comment (jeromtom) [reply CHANGELOG.md:16]: Okay, reverted the changelog entry. This PR might only have this URL fix as I have doubts regarding the rest.

**Comment by jeromtom** [comment]
> Thanks, @jeromtom! Will you be pushing more URL fixes to this PR?

Hey @Saransh-cpp,
The installation link (internal) and all external links as mentioned in #2660 have been updated in this PR.

The file/location where the broken links are located can be found from [Google Sheet-PyBaMM Link Fixes](https://docs.google.com/spreadsheets/d/16ZapQc9CE0XzdqT_ck_mtRR1jOxSDoE27oRlWMNEKfM/edit?usp=sharing)

For the other internal links, I think is best if we first create [the master documentation website](https://github.com/pybamm-team/PyBaMM/wiki/GSoC-2023-Projects#documentation) and then update the links.

**Comment by jeromtom** [comment]
@Saransh-cpp, I have committed all the correct links that I know of. There are broken links whose correct link I don't know, which are blank in [Google Sheet-PyBaMM Link Fixes.](https://docs.google.com/spreadsheets/d/16ZapQc9CE0XzdqT_ck_mtRR1jOxSDoE27oRlWMNEKfM/edit?usp=sharing)

**Comment by jeromtom** [comment]
@Saransh-cpp @kinnala @martinjrobins @bessman, kindly review this PR.

File: docs/index.rst [hunk 2]
```diff
@@ -19,11 +19,11 @@ PyBaMM documentation
 **Version**: |version|
 
 .. **Download documentation**:
-.. `PDF Version <https://numpy.org/doc/stable/numpy-user.pdf>`_ |
+.. `PDF Version <https://numpy.org/doc/1.23/numpy-user.pdf>`_ |
```
Comment (bessman) [inline docs/index.rst:22]: Why link to numpy's documentation at all? This looks like a copy-paste error that got introduced in 14abfe4b.

**Review by bessman — COMMENTED** [review]
Not sure why I'm being asked to weigh in. I guess I touched one of these files at some point?

**Comment by tinosulzer** [comment]
@jeromtom please make the changes suggested. Other than that it looks good, though the link checker didn't run so there are probably still some broken links (but fewer than before)

**Comment by jeromtom** [comment]
When I was considering who to mention, GitHub suggested a few names and I tagged them. Now realize that was not the way to go about it.
image

**Comment by jeromtom** [comment]
@tinosulzer, I have made the suggested changes. Let me know if anything else is required.

**Comment by tinosulzer** [comment]
You need to merge develop and fix the conflicts

**Comment by jeromtom** [comment]
The conflict was that I did not have permission to edit README.md which contained a broken link, hence reverted the commit resolving the conflict. @tinosulzer, ready for review.

**Review by tinosulzer — APPROVED** [review]

**Comment by tinosulzer** [comment]
@all-contributors add @jeromtom for docs

**Pull request merged**